src/DGtal/kernel/PointVector.ih

```diff
@@ -665,6 +665,22 @@ negate()
 template<DGtal::Dimension dim, typename TComponent, typename TContainer>
 inline
 double
+DGtal::PointVector<dim, TComponent, TContainer>::sqnorm () const
+{
+  double tmp = 0.0;
+  ASSERT ( dim > 0 );
+  double v;
+  for ( DGtal::Dimension i = 0; i < dimension; i++ )
+  {
+    v =NumberTraits<Component>::castToDouble(myArray[ i ]);
+    tmp += v*v;
```

Comment:
Why not simply `tmp += std::pow( NumberTraits<Component>::castToDouble(myArray[ i ]), 2);` ?

Reply:
well.. for performances.. 
I expect that all implementations of `std::pow(...,2)` to be specialized and optimized  but I'm not sure so I prefer the code to be explicit.

for exponents !=2, I would have used the `std::pow`

Reply:
I think it should be optimized by any decent compiler (see https://godbolt.org/z/TYR7wA) but ok :wink:

Reply:
:)

Reply:
@dcoeurjo I was partly mistaken on that point: I knew that since C++11, all `std::pow` specializations (e.g. for `double^int`) were grouped into (almost) one remaining definition (see (7) in https://en.cppreference.com/w/cpp/numeric/math/pow) but I only realized very recently that all integral parameters were converted to floating-point type before applying the power operator (see the description of (7) on the same link).

Thus, when calculating `std::pow(v, 3)` with `v` a double, the exponent 3 is stored as a double (or converted to a double if unknown) before calling (and not inlining) the (thus general) `pow` function (see https://godbolt.org/z/3MwAqz and compare the result when using `-std=c++98`).

If working with `float`, things get worst: there exists a `std::pow(float, float)` overload (item (1)) but probably because passing the integer 3 as the exponent involves a conversion, the compiler choose the overload (7) (that use template parameters) and that leads to `float <-> double` conversions before and after calling the `pow` function (see https://godbolt.org/z/Rx-bvo). The right overload can be forced by passing `3.f` as exponent or by explicitly using the `std::powf` function.

However, it seems that with an exponent of 2, the compilers still inline the `pow` function without  spurious conversions but I don't know if it is in the standard...

PS: If you wonder about vectorization, look at https://godbolt.org/z/HRcsRW ...